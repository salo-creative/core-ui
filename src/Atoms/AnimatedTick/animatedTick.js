  
import React from 'react';
import { delay } from 'lodash';

// STYLES
import { Wrapper, Path, PolyLine } from './animatedTick.styles';

class AnimatedTick extends React.Component {
  state = { show: false }

  componentDidMount = () => {
    delay(this.animateTick, 1);
  }

  animateTick = () => {
    const { show } = this.state;
    this.setState({ show: !show });
  }

  render() {
    const { show } = this.state;
    return (
      <Wrapper className={ show ? 'active' : '' }>
        <svg
          x='0px'
          y='0px'
          viewBox='0 0 37 37'
          style={ { 'enableBackground': 'new 0 0 37 37' } }
          xmlSpace='preserve'
        >
          <Path d='M30.5,6.5L30.5,6.5c6.6,6.6,6.6,17.4,0,24l0,0c-6.6,6.6-17.4,6.6-24,0l0,0c-6.6-6.6-6.6-17.4,0-24l0,0C13.1-0.2,23.9-0.2,30.5,6.5z' />
          <PolyLine points=' 11.6,20 15.9,24.2 26.4,13.8 ' />
        </svg>
      </Wrapper>
    );
  }
}

export default AnimatedTick;