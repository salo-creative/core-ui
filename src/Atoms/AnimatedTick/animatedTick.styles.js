
import styled from 'styled-components';

export const Wrapper = styled.div`
  margin: 0 auto;
  max-width: 40px;
  width: 100%;
`;

export const Path = styled.path`
  fill: none;
  stroke: ${ ({ theme }) => theme.success };
  stroke-width: 3;
  stroke-linejoin: round;
  stroke-miterlimit: 10;
  opacity: 0;
  stroke-dasharray: 130;
  stroke-dashoffset: 130;
  transition: all 1s;
  .active & {
    opacity: 1;
    stroke-dashoffset: 0;
  }
`;

export const PolyLine = styled.polyline`
  fill: none;
  stroke: ${ ({ theme }) => theme.success };
  stroke-width: 3;
  stroke-linejoin: round;
  stroke-miterlimit: 10;
  stroke-dasharray: 40;
  stroke-dashoffset: 40;
  transition: stroke-dashoffset 1s 0.5s ease-out;
  .active & {
    stroke-dashoffset: 0;
  }
`;