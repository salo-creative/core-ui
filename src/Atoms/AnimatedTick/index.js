import AnimatedTick from './animatedTick';

export default AnimatedTick;