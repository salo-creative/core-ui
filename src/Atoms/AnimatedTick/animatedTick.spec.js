import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../test';

// COMPONENT
import AnimatedTick from './index';

test('Renders against snapshot', async () => {
  const { container } = renderWithTheme(
    <AnimatedTick />
  );
  expect(container.firstChild).toMatchSnapshot();
});