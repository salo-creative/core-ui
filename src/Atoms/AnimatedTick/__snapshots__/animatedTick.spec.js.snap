// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders against snapshot 1`] = `
.c0 {
  margin: 0 auto;
  max-width: 40px;
  width: 100%;
}

.c1 {
  fill: none;
  stroke: #31D238;
  stroke-width: 3;
  stroke-linejoin: round;
  stroke-miterlimit: 10;
  opacity: 0;
  stroke-dasharray: 130;
  stroke-dashoffset: 130;
  -webkit-transition: all 1s;
  transition: all 1s;
}

.active .c1 {
  opacity: 1;
  stroke-dashoffset: 0;
}

.c2 {
  fill: none;
  stroke: #31D238;
  stroke-width: 3;
  stroke-linejoin: round;
  stroke-miterlimit: 10;
  stroke-dasharray: 40;
  stroke-dashoffset: 40;
  -webkit-transition: stroke-dashoffset 1s 0.5s ease-out;
  transition: stroke-dashoffset 1s 0.5s ease-out;
}

.active .c2 {
  stroke-dashoffset: 0;
}

<div
  class="c0"
>
  <svg
    style="enable-background: new 0 0 37 37;"
    viewBox="0 0 37 37"
    x="0px"
    xml:space="preserve"
    y="0px"
  >
    <path
      class="c1"
      d="M30.5,6.5L30.5,6.5c6.6,6.6,6.6,17.4,0,24l0,0c-6.6,6.6-17.4,6.6-24,0l0,0c-6.6-6.6-6.6-17.4,0-24l0,0C13.1-0.2,23.9-0.2,30.5,6.5z"
    />
    <polyline
      class="c2"
      points=" 11.6,20 15.9,24.2 26.4,13.8 "
    />
  </svg>
</div>
`;
